src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -385,14 +387,14 @@ public BitArray RightShift(int count)
                             int left = m_array[++fromIndex] << (BitsPerInt32 - shiftCount);
                             m_array[toIndex++] = left | (int)right;
                         }
-                        uint mask = uint.MaxValue >> (BitsPerInt32 - m_length % BitsPerInt32);
+                        uint mask = uint.MaxValue >> (BitsPerInt32 - extraBits);
```

Comment:
Optimized:

```cs
uint mask = (1u << extraBits) - 1u;
```

Reply:
This change results in test failures.

Reply:
I have added a comment explaining why we can't do this, based on our discussion. Thanks.